src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -5,62 +5,52 @@
 
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.IO;
+using System.Linq;
+using System.Reflection;
 using System.Runtime.InteropServices;
 using System.Text.RegularExpressions;
 
+#nullable enable
+
 namespace Microsoft.Build.Locator
 {
     internal static class DotNetSdkLocationHelper
     {
-        private static readonly Regex DotNetBasePathRegex = new Regex("Base Path:(.*)$", RegexOptions.Multiline);
         private static readonly Regex VersionRegex = new Regex(@"^(\d+)\.(\d+)\.(\d+)", RegexOptions.Multiline);
-        private static readonly Regex SdkRegex = new Regex(@"(\S+) \[(.*?)]$", RegexOptions.Multiline);
         private static readonly bool IsWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        private static readonly string ExeName = IsWindows ? "dotnet.exe" : "dotnet";
+        private static readonly string DotnetPath = ResolveDotnetPath();
 
-        public static VisualStudioInstance GetInstance(string dotNetSdkPath)
-        {            
-            if (string.IsNullOrWhiteSpace(dotNetSdkPath))
-            {
-                return null;
-            }
+        static DotNetSdkLocationHelper() => LoadHostFxr();
```

Comment:
Why do these methods run as part of static/type initialization? It would be better to run them when actually needed, one reason is exception handling where we would prefer determinism as to when exceptions are thrown.

Reply:
In order to run LoadHostFxr(), we need to Resolve dotnet path. Nathan pointed out in his comment that we could potentially fix other hostfxr-related problems
https://github.com/microsoft/MSBuildLocator/pull/230#discussion_r1302149135

Reply:
If you think it's not our case, I can move it back in the method invocation.

Reply:
I'm not sure I understand the comment. This is an internal class and we control the order of method invocations. Doing this work in the static constructor would be bad practice.

Reply:
`ResolveDotnetPath` is the more interesting one. I have left a comment to move this off of the static constructor also.